To: vim_dev@googlegroups.com
Subject: Patch 7.3.142
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.142
Problem:    Python stdout doesn't have a flush() method, causing an import to
	    fail.
Solution:   Add a dummy flush() method. (Tobias Columbus)
Files:	    src/if_py_both.h


*** ../vim-7.3.141/src/if_py_both.h	2010-10-27 12:17:54.000000000 +0200
--- src/if_py_both.h	2011-03-22 15:37:43.000000000 +0100
***************
*** 33,38 ****
--- 33,39 ----
  
  static PyObject *OutputWrite(PyObject *, PyObject *);
  static PyObject *OutputWritelines(PyObject *, PyObject *);
+ static PyObject *OutputFlush(PyObject *, PyObject *);
  
  /* Function to write a line, points to either msg() or emsg(). */
  typedef void (*writefn)(char_u *);
***************
*** 47,55 ****
  
  static struct PyMethodDef OutputMethods[] = {
      /* name,	    function,		calling,    documentation */
!     {"write",	    OutputWrite,	1,	    "" },
!     {"writelines",  OutputWritelines,	1,	    "" },
!     { NULL,	    NULL,		0,	    NULL }
  };
  
  #define PyErr_SetVim(str) PyErr_SetString(VimError, str)
--- 48,57 ----
  
  static struct PyMethodDef OutputMethods[] = {
      /* name,	    function,		calling,    documentation */
!     {"write",	    OutputWrite,	1,	    ""},
!     {"writelines",  OutputWritelines,	1,	    ""},
!     {"flush",       OutputFlush,        1,          ""},
!     { NULL,	    NULL,		0,	    NULL}
  };
  
  #define PyErr_SetVim(str) PyErr_SetString(VimError, str)
***************
*** 123,128 ****
--- 125,139 ----
      return Py_None;
  }
  
+     static PyObject *
+ OutputFlush(PyObject *self UNUSED, PyObject *args UNUSED)
+ {
+     /* do nothing */
+     Py_INCREF(Py_None);
+     return Py_None;
+ }
+ 
+ 
  /* Buffer IO, we write one whole line at a time. */
  static garray_T io_ga = {0, 0, 1, 80, NULL};
  static writefn old_fn = NULL;
*** ../vim-7.3.141/src/version.c	2011-03-22 14:35:01.000000000 +0100
--- src/version.c	2011-03-22 15:45:38.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     142,
  /**/

-- 
Living on Earth includes an annual free trip around the Sun.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
